pydantic/mypy.py

```diff
@@ -608,6 +657,10 @@ def error_untyped_fields(api: SemanticAnalyzerPluginInterface, context: Context)
     api.fail('Untyped fields disallowed', context, code=ERROR_UNTYPED)
 
 
+def error_default_and_default_factory_specified(api: CheckerPluginInterface, context: Context) -> None:
```

Comment:
is there a good reason to make this a standalone function?

Reply:
The main reason is that all the `api.fail` are wrapped into functions for clarity, so I thought of being consistent with the rest of the codebase. What do you think?

Reply:
sounds good.